documentation/design/user-stories/shared-infra.adoc

```diff
@@ -182,17 +163,71 @@ metadata:
   name: endpoint1
   namespace: myapp
 spec:
-  # TLS configuration for endpoint
-  tls: {}
-
   protocols:
-    - AMQPS
-    - AMQP
+  - AMQP
+  - AMQPS
+  - AMQP-WS
+  - AMQP-WSS
 
+  # Optional: Override generated hostname for endpoint. Clients must have the AMQP OPEN hostname set to this value.
```

Comment:
Will this be ignored for endpoints except route and ingress?

Reply:
It is useful for both loadbalancer and nodeport as well, because the router will expect that host for clients connecting. So if you want to setup messaging on messaging.example.com, you create an endpoint with that hostname, and then point your DNS record to the created loadbalancer.

For nodeport and local development, the host would be set to something like $(minikube ip) for clients to be able to connect outside the cluster.

For cluster service, it is less useful, but you could end up setting the AMQP host in your client and connect to the ip instead of the host, and it would work.